### DISCUSSION TITLE
Allowing wallet_watchAsset and other wallet APIs

### DISCUSSION THREAD

**Discussion by alexshelkov** [MetaMask/snaps#2014] [Q&A] [answered]
Hey, 

I just want to understand is there is some roadmap or date when the UI for the wallet related API (like: `wallet_switchEthereumChain` `wallet_watchAsset`) is ready and they can be used in snaps? Maybe we have some planned release date for this features?

Thanks!

**Comment by Montoya** [comment, marked as answer]
Hi @alexshelkov. As of now we do not have a roadmap for supporting more wallet methods like `wallet_switchEthereumChain` or `wallet_watchAsset`. It's something we can explore in Q1. In the meantime, I'm curious to hear how you would use these methods!

**Reply by alexshelkov** [reply]
So I just want to create a snap which would allow to add custom network and add the tokens for this network to the wallet.

**Reply by Montoya** [reply]
OK that makes sense! We are exploring how the Snaps API can make this possible in the future. I will have to get back to you when we make more progress on this in Q1 '24.

**Reply by patcito** [reply]
hey @Montoya, I would be interested in having wallet_switchEthereumChain supported too. We provided an RPC endpoint with mev protection and we would want to allow users to switch to our RPC endpoint on tx confirmation dialog.
